File: toy/e-commerce-java/src/main/java/me/kzv/ecommerce/repository/ItemImgRepository.java
package me.kzv.ecommerce.repository;

import me.kzv.ecommerce.entity.ItemImg;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface ItemImgRepository extends JpaRepository<ItemImg, Long> {
    List<ItemImg> findByItemIdOrderByIdAsc(Long itemId); // fk 인 itemId 로 검색

    ItemImg findByItemIdAndRepimgYn(Long itemId, String repimgYn);
}

File: toy/e-commerce/src/main/kotlin/me/kzv/shop/repository/ItemImgRepository.kt
package me.kzv.shop.repository

import me.kzv.shop.entity.ItemImg
import org.springframework.data.jpa.repository.JpaRepository

interface ItemImgRepository : JpaRepository<ItemImg, Long> {

    fun findByItemIdOrderByIdAsc(itemId: Long): List<ItemImg>

    fun findByItemIdAndMainImgYn(itemId: Long, mainImgYn: String)
}